app/javascript/react/components/AddMonsterTile.js

```diff
@@ -0,0 +1,14 @@
+import React from "react"
+import { Link } from "react-router-dom"
+
+const AddMonsterTile = props => {
+  const {signedIn} = props
+
+  if(signedIn) {
+    return <Link to="/monsters/new">Add Monster</Link>
+  } else {
+    return <a href="/users/sign_in">Add Monster</a>
+  }
+}
```

Comment:
okay, you get points for being very crafty. However, this component is starting to have very clearly different responsibilities. 

I will allow this, but I personally believe it would be better to have NOT made this into a component at all, and simply had your logic in your navvar